Sources/PackageModel/Manifest/PackageDependencyDescription.swift

```diff
@@ -11,15 +11,33 @@
 import TSCBasic
 
 /// Represents a package dependency.
-public struct PackageDependencyDescription: Equatable, Codable, Hashable {
+public enum PackageDependencyDescription: Equatable {
+
+     public struct Local: Equatable, Codable {
+        public let identity: PackageIdentity
+        public let name: String?
+        public let path: AbsolutePath
+        public let productFilter: ProductFilter
+    }
+
+    public struct Git: Equatable, Codable {
+        public let identity: PackageIdentity
+        public let name: String?
+        public let location: String
+        public let requirement: Requirement
+        public let productFilter: ProductFilter
+    }
+
+    case local(Local)
+    case git(Git)
+    //case registry(data: Registry) // for future
 
     /// The dependency requirement.
     public enum Requirement: Equatable, Hashable {
         case exact(Version)
         case range(Range<Version>)
         case revision(String)
         case branch(String)
-        case localPackage
```

Comment:
a nice side effect of the move to enum is that we can get rid of `localPackage` as kind of requirement which seems unnatural

Reply:
Agreed, that was always odd.